### DISCUSSION TITLE
Initialization of ZIP

### DISCUSSION THREAD

**Discussion by xiaoheimai95** [CURENT/andes#459] [Modeling] [answered]
Dear Dr. Cui,
We had some confusion when referring to the guide to initialize the ZIP device. In the guide, it states that ZIP model should be initialized after power flow and should set p2p=1.0 and q2q=1.0, (p2i, p2z, q2i, q2z, and pq2z) to zero. However, in examples 2.9.1 and 2.9.2, these formulas are also involved, but they are all changed before the power flow. So where exactly should we convert them when we wanna use ZIP?
![屏幕截图 2023-05-31 174550](https://github.com/CURENT/andes/assets/71546641/59d26682-727d-4327-8b8b-c1d265609f00)
![屏幕截图 2023-05-31 174622](https://github.com/CURENT/andes/assets/71546641/e08aced8-a7fc-4cb9-bab1-8365b5f2da25)

**Comment by jinningwang** [comment]
If one is using ZIP, these settings "p2p=1.0 and q2q=1.0, (p2i, p2z, q2i, q2z, and pq2z)" are mandatory ***before*** running power flow. Otherwise, these settings are optional and depended on the users' preference. As stated in [ZIP](https://docs.andes.app/en/latest/groupdoc/DynLoad.html?highlight=ZIP#zip), ***This model is initialized after power flow.***

For the example you mentioned, are you referring to this [Tripping a Generator in the IEEE 14-Bus System](https://docs.andes.app/en/latest/_examples/ex9.html#tripping-a-generator-in-the-ieee-14-bus-system)? Here we also use these settings to make the load as ***constant load model*** to make sure the shed load values are equal to the missed generation.

Hope this addresses your concern.

**Reply by xiaoheimai95** [reply]
Dear Jinning,
I had some other issues last week, sorry for the late reply.
My concern is that, in the case 'tripping a generator in the IEEE 14-Bus system', the `ss.PFlow.run()` is used after the `p2p=1.0 and q2q=1.0, (p2i, p2z, q2i, q2z, and pq2z)` setting. So I am a little confused about that when I use the ZIP model, I should set the `p2p=1.0 and q2q=1.0, (p2i, p2z, q2i, q2z, and pq2z)` before the code  `ss.PFlow.run()`  or after it. Like:

```
ss.setup()
ss.PQ.config.p2p = 1.0
ss.PQ.config.q2q = 1.0
ss.PQ.config.p2i = 0
ss.PQ.config.p2z = 0
ss.PQ.config.q2i = 0
ss.PQ.config.q2z = 0
ss.PQ.config.pq2z = 0
ss.PFlow.run()
```
or:
```
ss.setup()
ss.PFlow.run()
ss.PQ.config.p2p = 1.0
ss.PQ.config.q2q = 1.0
ss.PQ.config.p2i = 0
ss.PQ.config.p2z = 0
ss.PQ.config.q2i = 0
ss.PQ.config.q2z = 0
ss.PQ.config.pq2z = 0
```
which type of the code should I use?

**Reply by cuihantao** [reply, marked as answer]
The first one will do.
```
ss.PQ.config.p2p = 1.0
ss.PQ.config.q2q = 1.0
ss.PQ.config.p2i = 0
ss.PQ.config.p2z = 0
ss.PQ.config.q2i = 0
ss.PQ.config.q2z = 0
ss.PQ.config.pq2z = 0
ss.PFlow.run()
```
sets config for the `PQ` model, so that it stays constant power.

The `ZIP` model is initialized after power flow using the constant power as set in the `PQ`.

**Reply by cuihantao** [reply]
To configure the ZIP load for simulation, you can change the parameters of ZIP directly in the data file.
